WordPress/openverse-catalog
# Re-ping if PR is updated and don't ping if 2 approvals exist

## Fixes


Fixes #571  by @sarayourfriend 

## Description


1. Fixes the formatting of the markdown of the comment and adds some additional explanation footnotes to the comment
2. Deletes previous review comments when leaving a re-ping to declutter the PR conversation history
3. Re-pings when the PR has been updated since the last reminder and the PR is due again
4. Does not ping if the PR already has two approvals

## Testing Instructions

Checkout this PR locally, add a GitHub PAT with access to the relevant repositories in your local Airflow variables under the name `GITHUB_API_KEY`. Run the dag in your local airflow and check that the dry run logs work. Note that right now I only see one reminder being generated, this is pretty much up to the state of our repositories so the unit tests should be the primary indication of whether this DAG is working properly.

Check out the new unit tests and make sure they make sense. Also let me know if any new ones need to be added.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
